### PR TITLE
Add support for dismissible notifications and convert existing alerts.

### PR DESCRIPTION
**Pull request by zrideaux**
+ Add alerts section to comments tab to hold UI error alerts.
+ Add createErrorAlert() and dismissErrorAlert() functions to add and remove new alerts to and from alerts section.
+ Add styles for UI alerts. 
+ Change existing instances of alert() calls to createErrorAlert() calls. 
+ Add front end check to ensure fields are filled. Call createErrorAlert() if they are not.

image

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -120,22 +120,26 @@ function addComment() {
...
 
-    http.onreadystatechange = function () {
-        if (http.readyState == 4 && http.status == 200) {
-            // Refresh comment section
-            getCommentsFromServlet();
+        http.onreadystatechange = function () {
+            if (http.readyState == 4 && http.status == 200) {
+                // Refresh comment section
+                getCommentsFromServlet();
+            }
```
Comment (saucepwn) [inline portfolio/src/main/webapp/script.js:138]: There's a good opportunity here to add an `} else if (http.status >= 400) { // Show HTTP error message }`

File: portfolio/src/main/webapp/style.css [hunk 2]
```diff
@@ -264,6 +265,26 @@ input {
     display: none;
 }
 
+.error-alert {
+  background-color: #E05571;
+  border-radius: 3px;
+  color: #FFFFFF;
+  display: flex;
+  justify-content: space-between;
+  margin: 5px 0px 5px 0px;
```
Comment (saucepwn) [inline portfolio/src/main/webapp/style.css:274]: You can shorthand this to:

```suggestion
  margin: 5px 0; 
```

ref) https://www.w3schools.com/cssref/pr_margin.asp

Comment (zrideaux) [reply portfolio/src/main/webapp/style.css:274]: Oh, awesome, didn't know about this! Done.

File: portfolio/src/main/webapp/script.js (hunk 1 above)
Comment (zrideaux) [reply portfolio/src/main/webapp/script.js:138]: True! Added

**Pull request merged**